src/server/service/diagnostics/log.rs

```diff
@@ -323,6 +346,8 @@ mod tests {
     use std::io::Write;
     use tempfile::tempdir;
 
+    const INVALID_TIMESTAMP: i64 = -1;
```

Comment:
I don't think this constant is needed. For the function at L382, it can just be:
```rust
        DateTime::parse_from_str(time, "%Y/%m/%d %H:%M:%S%.3f %z")
            .unwrap()
            .timestamp_millis()
```

Reply:
Ok, let me try it.

Reply:
Done!